workflow/cron/operator.go

```diff
@@ -309,6 +310,14 @@ func (woc *cronWfOperationCtx) enforceHistoryLimit(ctx context.Context, workflow
 				successfulWorkflows = append(successfulWorkflows, wf)
 			} else {
 				failedWorkflows = append(failedWorkflows, wf)
+				if woc.cronWf.Spec.SuspendAfterFailedJobsLimit != nil && *woc.cronWf.Spec.SuspendAfterFailedJobsLimit >= 0 {
+					errCount++
```

Comment:
A couple of options here:
1. Track this count within failed history limit
2. Track this count globally which goes beyond the history limit

Reply:
What happens if we get submission errors and workflow objects are never created?

Reply:
Or if workflows have GC on them and are deleted immediately after failure?